components-compose/src/main/java/uk/gov/hmrc/components/compose/molecule/input/PasswordInputView.kt

```diff
@@ -0,0 +1,376 @@
+        ),
+        interactionSource = interactionSource,
+        requiredSequencesSpacing = requiredSequencesSpacing,
+        textStyle = textStyle
+    )
+}
+
+@Composable
+fun animateBorderStrokeAsState(
+    enabled: Boolean,
+    isError: Boolean,
+    interactionSource: InteractionSource,
+    focusedBorderThickness: Dp,
+    unfocusedBorderThickness: Dp
+): State<BorderStroke> {
+    val focused by interactionSource.collectIsFocusedAsState()
+    val indicatorColor = indicatorColor(enabled, isError, interactionSource)
+    val targetThickness = if (focused) focusedBorderThickness else unfocusedBorderThickness
+    val animatedThickness = if (enabled) {
+        animateDpAsState(targetThickness, tween(durationMillis = ANIMATION_DURATION))
+    } else {
+        rememberUpdatedState(unfocusedBorderThickness)
+    }
+    return rememberUpdatedState(
+        BorderStroke(animatedThickness.value, SolidColor(indicatorColor.value))
+    )
+}
+
+var borderStroke: State<BorderStroke>? = null
+
+@Composable
+fun CalculateBorder(
+    enabled: Boolean = true,
+    isError: Boolean,
+    interactionSource: InteractionSource,
+    focusedBorderThickness: Dp = FocusedBorderThickness,
+    unfocusedBorderThickness: Dp = UnfocusedBorderThickness
+) {
+    borderStroke = animateBorderStrokeAsState(
+        enabled,
+        isError,
+        interactionSource,
+        focusedBorderThickness,
+        unfocusedBorderThickness
+    )
+}
+
+@Composable
+fun indicatorColor(
+    enabled: Boolean,
+    isError: Boolean,
+    interactionSource: InteractionSource
+): State<Color> {
+    val focused by interactionSource.collectIsFocusedAsState()
+
+    val colors = HmrcTheme.colors
+
+    val focusedBorderColor = colors.hmrcBlue
+    val errorBorderColor = colors.hmrcBlack
+    val unfocusedBorderColor = colors.hmrcBlack
+
+    val targetValue = when {
+        isError -> errorBorderColor
+        focused -> focusedBorderColor
+        else -> unfocusedBorderColor
+    }
+    return if (enabled) {
+        animateColorAsState(targetValue, tween(durationMillis = ANIMATION_DURATION))
+    } else {
+        rememberUpdatedState(targetValue)
+    }
+}
+
+@SuppressWarnings("LongMethod", "ComplexMethod")
+@Composable
+fun PasswordTextInputView(
+    modifier: Modifier = Modifier,
+    value: String? = null,
+    onInputValueChange: ((String) -> Unit)? = null,
+    inputFilter: ((String, String) -> String)? = null,
+    labelText: String? = null,
+    labelContentDescription: String? = null,
+    hintText: String? = null,
+    hintContentDescription: String? = null,
+    prefix: @Composable() (() -> Unit)? = null,
+    placeholderText: String? = null,
+    errorText: String? = null,
+    errorContentDescription: String? = null,
+    showButtonText: String? = stringResource(R.string.passwordTextInputView_show),
+    hideButtonText: String? = stringResource(R.string.passwordTextInputView_hide),
```

Comment:
I think it would be safer to not provide default parameters here, as we want the consumer to provide localised versions of the strings.